### ISSUE TITLE
Missing 'CHUNK_EXTRA' causes the node to stall

### ISSUE THREAD

**Issue by mm-near** [near/nearcore#6442]
Happened on localnet - after restarting a job it could no longer catch up.

When I've enabled debug logs, I've noticed:

 Mar 16 11:24:08.157 DEBUG chain: Process block header: CNJS5wi12zjN5o91H5kSK1YMkeSfZDnaG8UFSvDrfzww at 12676
 Mar 16 11:24:08.158 DEBUG Process block

 {#12676}: chain: enter

 Mar 16 11:24:08.158 DEBUG Process block{#12676} 

: chain: Block CNJS5wi12zjN5o91H5kSK1YMkeSfZDnaG8UFSvDrfzww, approvals: 2, me: Some(AccountId("node0"))
 Mar 16 11:24:08.158 DEBUG Process block

 {#12676}: chain: Some(AccountId("node0")) Process block CNJS5wi12zjN5o91H5kSK1YMkeSfZDnaG8UFSvDrfzww, is_caught_up: true, need_to_start_fetching_state: false

 Mar 16 11:24:08.159 DEBUG Process block{#12676} 

: chain: Verifying challenges []
 Mar 16 11:24:08.159 DEBUG Process block

 {#12676} 

: chain: close time.busy=985µs time.idle=3.43µs
 Mar 16 11:24:08.159 DEBUG client: Error on receival of block: DB Not Found Error: CHUNK EXTRA: A9qwvxpwkBJoT39BJ5NyQwWJv3f18agRh14hXFZSEHRa:ShardUId 

 { version: 1, shard_id: 0 } 

The A9.. block the block 12675 which looks to be 'successfully' applied on this node. So for some reason we didn't persist the chunk extra, and now the node is fully stuck.

As the first step - we should consider printing some error message - to explain more to the user on what's going on.

**Comment by mm-near** [comment]
this seems to happen - if you were a validator, then you got kicked out (which means that by default you're not tracking any shards) - and then you try to track some shards.

In such case, when you change the config (to add new tracked shards) and restart - the node will get stuck with the message above.

**Comment by bowenwang1996** [comment]
> In such case, when you change the config (to add new tracked shards) and restart - the node will get stuck with the message above.

Yes it is known that changing tracked shards may cause a node to break. I think we should just shutdown the node in this case to prevent further damage. Wouldn't say it is bug because it is designed to be this way from the very beginning
